test/test_real_space.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+
+from pseudospectral import DiracOperator, Derivative1D
+import numpy as np
+import pytest
+
+def test_application_to_a_single_eigenfunction(L=4, n=4):
+    """
+    Python test function to test the application of the Dirac operator to a single eigenfunction in real space.
+    """
+    arbitrary_index = 1 # Index of the eigenfunction to be tested
+    operator = DiracOperator(Derivative1D(num_lattice_points=n, L=L))
+    
+    eigenfunction = operator.spectrum.eigenfunction(arbitrary_index)(np.arange(L))
```

Comment:
```suggestion
    eigenfunction = operator.spectrum.eigenfunction(arbitrary_index)(np.arange(n)/n*L)
```
You're missing your length parameter here. This is why I advocated for having it set to one in the first place! Please consider again to remove this feature for now. We can add it back when we need it.

Scrolling through the rest I see that change must, of course, be made everywhere.

Reply:
I set n=L as you suggested. So therefore it boils down to my formula

Reply:
Well, the important thing is not so much if it works (right now) or not but if it is semantically correct. And this isn't...

But on a more practical note, I'm aware that this test works for the default parameters. But you've made them function arguments, so someone could come along and pass something else. Then, this would be wrong.